### DISCUSSION TITLE
Loading custom asset

### DISCUSSION THREAD

**Discussion by joetsoi** [a1phyr/ggez-assets_manager#1] [Q&A] [answered]
If I've implemented `Asset` for a custom type, with a custom loader, how should I translate the ggez filename to an asset id?
```rust
// standard ggez 0.8
let mut file = ctx.fs.open(Path::new("/what/goes/here.myfile")).unwrap();
let background = MyType::from_reader(&mut file).unwrap();

// how does translate to asset_manager?
let handle = assets.load::<MyType>("what.goes.here").unwrap();
```

**Comment by a1phyr** [comment, marked as answer]
This is exactly as you've shown it. 

Additionally, the extension is set with either `EXTENSION` or `EXTENSIONS` in `Asset` trait (cf https://docs.rs/assets_manager/latest/assets_manager/asset/trait.Asset.html#provided-associated-consts)

**Reply by joetsoi** [reply]
Ah! That works for a `resources` directory in the same level as the executable. I guess this is because https://docs.rs/ggez-assets_manager/latest/src/ggez_assets_manager/source.rs.html#30 explicitly looks in resources directory.

I was expecting it to pickup the ~/.local/share directory as listed https://docs.rs/ggez/latest/ggez/filesystem/index.html

I think it might also not work for [renaming the resources directory name](https://docs.rs/ggez/latest/ggez/context/struct.ContextBuilder.html#method.resources_dir_name) or [adding additional resource paths](https://docs.rs/ggez/latest/ggez/context/struct.ContextBuilder.html#method.add_resource_path)

**Reply by a1phyr** [reply]
Reading assets from local directory was supposed to work but did not due to a shameless copy-paste : https://github.com/a1phyr/ggez-assets_manager/blob/e3b1e9c6cfddba18f7e0f9b6a0a1215e224c33e3/src/source.rs#L45-L68

I've just released a new version with a fix. Additionally, `GgezFileSystem` is now directly created from ggez' Context, which makes it work with directory renaming (with an API that did not exist in `ggez` 0.7).
However, supporting adding additional resource paths will require more work/new API from ggez.

**Reply by a1phyr** [reply]
Anyway, thanks for reporting this issue !
